### PR TITLE
#137 fixed throwing errors using TouchUserInputVariable...

### PR DESCRIPTION
**Pull request by IceTrooper**
... when no camera on scene

### CODE REVIEW

**Comment by IceTrooper** [comment]
#137

**Comment by soraphis** [comment]
Hm. I don't know enough about the Touch implementation.

The fix seems to be okay, but I think that: 

`public Vector2 InputWorldPos { get { return Camera.main.ScreenToWorldPoint(InputPos); } }`

needs to be changed. Camera.main is an expensive method:

> This property uses FindGameObjectsWithTag internally and doesn't cache the result. It is advised to cache the return value of Camera.main if it is used multiple times per frame.
https://docs.unity3d.com/ScriptReference/Camera-main.html

also, I think if `Camera.main == null` all `WorldPos` should return some kind of zero (e.g. Vector2.zero) Instead of throwing an exception. (but this is up to discussion)

**Comment by IceTrooper** [comment]
Yes, you're right. I also think that something should change in InputWorldPosition property. It should return Vector2.zero if there isn't a camera on the scene. Based on @soraphis review I did some changes.

**Comment by soraphis** [comment]
API has been changed. those are breaking changes and should be put in the upcoming changes for `UnityAtoms v5`

File: Packages/Mobile/Runtime/Types/TouchUserInput.cs [hunk 1]
```diff
@@ -24,54 +24,73 @@ public enum State
         /// <summary>
         /// Current input state.
         /// </summary>
-        public State InputState;
+        public State TouchState { get => touchState; set => touchState = value; }
+        [SerializeField] private State touchState;
```
Comment (AdamRamberg) [inline Packages/Mobile/Runtime/Types/TouchUserInput.cs:28]: Use underscore as prefix. Eg. `_touchState`

File: Packages/Mobile/Runtime/Types/TouchUserInput.cs [hunk 2]
```diff
@@ -24,54 +24,73 @@ public enum State
...
 
         /// <summary>
         /// Current input position.
         /// </summary>
-        public Vector2 InputPos;
+        public Vector2 Position { get => position; set => position = value; }
+        [SerializeField] private Vector2 position;
```
Comment (AdamRamberg) [inline Packages/Mobile/Runtime/Types/TouchUserInput.cs:34]: Use underscore as prefix. Eg. `_position`

File: Packages/Mobile/Runtime/Types/TouchUserInput.cs [hunk 3]
```diff
@@ -24,54 +24,73 @@ public enum State
...
 
         /// <summary>
         /// Input position last frame.
         /// </summary>
-        public Vector2 InputPosLastFrame;
+        public Vector2 PositionLastFrame { get => positionLastFrame; set => positionLastFrame = value; }
+        [SerializeField] private Vector2 positionLastFrame;
```
Comment (AdamRamberg) [inline Packages/Mobile/Runtime/Types/TouchUserInput.cs:40]: Use underscore as prefix. Eg. `_positionLastFrame`

File: Packages/Mobile/Runtime/Types/TouchUserInput.cs [hunk 4]
```diff
@@ -24,54 +24,73 @@ public enum State
...
 
         /// <summary>
         /// Input position last time the user pressed down.
         /// </summary>
-        public Vector2 InputPosLastDown;
+        public Vector2 PositionLastDown { get => positionLastDown; set => positionLastDown = value; }
+        [SerializeField] private Vector2 positionLastDown;
```
Comment (AdamRamberg) [inline Packages/Mobile/Runtime/Types/TouchUserInput.cs:46]: Use underscore as prefix. Eg. `_positionLastDown`

**Comment by AdamRamberg** [comment]
Thanks for the PR! Like @soraphis said, this is a breaking change and we will need to wait adding this to Atoms until v5. FYI, the release of v5 of Atoms might take some time. The change for fixing the `Camera.main` issue could have been done in a none breaking change. So if you need this fix soon I would suggest 2 PRs, when for fixing `Camera.main` and one for do the refactoring of the prop / variable names etc. If no hurry, let the PR remain and we will add it to v5. 

Other than that I added some minior comments regarding this PR that needs to be fixed. Also created a new branch for v5 stuff that this PR will need to be merged to.

**Comment by AdamRamberg** [comment]
@IceTrooper How is progress going on implementing the style fixes? When done I can merge this into the v5 branch.

**Comment by IceTrooper** [comment]
Hey, there was a time I was busy, now I can make some changes. Tomorrow I will check it :)

**Comment by soraphis** [comment]
since serialized fields were renamed, shouldn't we add [FormerlySerializedAsAttribute](https://docs.unity3d.com/ScriptReference/Serialization.FormerlySerializedAsAttribute.html) to them?

**Comment by AdamRamberg** [comment]
@soraphis Yes, or only include this PR in v5, so merge it to the v5 branch instead of the canary branch.

**Pull request merged**